### Notebook 1

This notebook repoduces the experiments of the paper on the SHU-MI dataset. " SHU-MI is an EEG motor imagery dataset with two categories (left hand and right hand). It consists of 32-channel EEG signals with 250 Hz sampling rate, collected from 25 subjects. The EEG signals are segmented into 11,988 samples and resampled to 200 Hz. "

The fine-tuning function is already implemented in the repository, the default parameters being the one reported in the paper.

In [1]:
%run finetune_main.py \
    --downstream_dataset SHU-MI \
    --datasets_dir custom_shu/ \
    --num_of_classes 2 \
    --model_dir checkpoints \
    --epochs 50

Namespace(seed=3407, cuda=0, epochs=50, batch_size=64, lr=0.0001, weight_decay=0.05, optimizer='AdamW', clip_value=1, dropout=0.1, classifier='all_patch_reps', downstream_dataset='SHU-MI', datasets_dir='custom_shu/', num_of_classes=2, model_dir='checkpoints', num_workers=16, label_smoothing=0.1, multi_lr=True, frozen=False, use_pretrained_weights=True, foundation_dir='pretrained_weights/pretrained_weights.pth')
The downstream dataset is SHU-MI
7210 2431 2347
11988


Model(
  (backbone): CBraMod(
    (patch_embedding): PatchEmbedding(
      (positional_encoding): Sequential(
        (0): Conv2d(200, 200, kernel_size=(19, 7), stride=(1, 1), padding=(9, 3), groups=200)
      )
      (proj_in): Sequential(
        (0): Conv2d(1, 25, kernel_size=(1, 49), stride=(1, 25), padding=(0, 24))
        (1): GroupNorm(5, 25, eps=1e-05, affine=True)
        (2): GELU(approximate='none')
        (3): Conv2d(25, 25, kernel_size=(1, 3), stride=(1, 1), padding=(0, 1))
        (4): GroupNorm(5, 25, eps=1e-05, affine=True)
        (5): GELU(approximate='none')
        (6): Conv2d(25, 25, kernel_size=(1, 3), stride=(1, 1), padding=(0, 1))
        (7): GroupNorm(5, 25, eps=1e-05, affine=True)
        (8): GELU(approximate='none')
      )
      (spectral_proj): Sequential(
        (0): Linear(in_features=101, out_features=200, bias=True)
        (1): Dropout(p=0.1, inplace=False)
      )
    )
    (encoder): TransformerEncoder(
      (layers): ModuleList(
        (0-11):

  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.71it/s]

 68%|██████▊   | 77/113 [00:20<00:09,  3.80it/s]

100%|██████████| 113/113 [00:29<00:00,  3.82it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.69it/s]

 97%|█████████▋| 37/38 [00:02<00:00, 17.88it/s]

100%|██████████| 38/38 [00:02<00:00, 17.83it/s]

Epoch 1 : Training Loss: 0.70184, acc: 0.64310, pr_auc: 0.72686, roc_auc: 0.71063, LR: 0.00010, Time elapsed 0.53 mins
[[844 368]
 [500 719]]
roc_auc increasing....saving weights !! 
Val Evaluation: acc: 0.64310, pr_auc: 0.72686, roc_auc: 0.71063


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.79it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.79it/s]

100%|██████████| 113/113 [00:29<00:00,  3.80it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.84it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.89it/s]

100%|██████████| 38/38 [00:02<00:00, 17.88it/s]

Epoch 2 : Training Loss: 0.64955, acc: 0.64215, pr_auc: 0.72820, roc_auc: 0.72177, LR: 0.00010, Time elapsed 0.53 mins
[[615 597]
 [272 947]]
roc_auc increasing....saving weights !! 
Val Evaluation: acc: 0.64215, pr_auc: 0.72820, roc_auc: 0.72177


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.76it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.75it/s]

100%|██████████| 113/113 [00:30<00:00,  3.76it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.89it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.82it/s]

100%|██████████| 38/38 [00:02<00:00, 17.82it/s]

Epoch 3 : Training Loss: 0.61735, acc: 0.65608, pr_auc: 0.74123, roc_auc: 0.73509, LR: 0.00010, Time elapsed 0.54 mins
[[784 428]
 [408 811]]
roc_auc increasing....saving weights !! 
Val Evaluation: acc: 0.65608, pr_auc: 0.74123, roc_auc: 0.73509


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.72it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.72it/s]

100%|██████████| 113/113 [00:30<00:00,  3.73it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.76it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.57it/s]

100%|██████████| 38/38 [00:02<00:00, 17.59it/s]

Epoch 4 : Training Loss: 0.58276, acc: 0.66117, pr_auc: 0.75803, roc_auc: 0.74305, LR: 0.00010, Time elapsed 0.54 mins
[[856 356]
 [468 751]]
roc_auc increasing....saving weights !! 
Val Evaluation: acc: 0.66117, pr_auc: 0.75803, roc_auc: 0.74305


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.73it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.72it/s]

100%|██████████| 113/113 [00:30<00:00,  3.74it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.84it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.76it/s]

100%|██████████| 38/38 [00:02<00:00, 17.77it/s]

Epoch 5 : Training Loss: 0.53074, acc: 0.65760, pr_auc: 0.75579, roc_auc: 0.74508, LR: 0.00010, Time elapsed 0.54 mins
[[907 305]
 [528 691]]
roc_auc increasing....saving weights !! 
Val Evaluation: acc: 0.65760, pr_auc: 0.75579, roc_auc: 0.74508


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.71it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.73it/s]

100%|██████████| 113/113 [00:30<00:00,  3.74it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.81it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.82it/s]

100%|██████████| 38/38 [00:02<00:00, 17.79it/s]

Epoch 6 : Training Loss: 0.45674, acc: 0.65979, pr_auc: 0.72099, roc_auc: 0.71404, LR: 0.00010, Time elapsed 0.54 mins
[[792 420]
 [407 812]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.74it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.75it/s]

100%|██████████| 113/113 [00:30<00:00,  3.76it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.96it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.82it/s]

100%|██████████| 38/38 [00:02<00:00, 17.82it/s]

Epoch 7 : Training Loss: 0.36264, acc: 0.62572, pr_auc: 0.71065, roc_auc: 0.69987, LR: 0.00010, Time elapsed 0.54 mins
[[781 431]
 [479 740]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.74it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.75it/s]

100%|██████████| 113/113 [00:30<00:00,  3.76it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.07it/s]

100%|██████████| 38/38 [00:02<00:00, 17.82it/s]

100%|██████████| 38/38 [00:02<00:00, 17.85it/s]

Epoch 8 : Training Loss: 0.26171, acc: 0.63341, pr_auc: 0.71051, roc_auc: 0.70663, LR: 0.00009, Time elapsed 0.54 mins
[[738 474]
 [417 802]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.76it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.77it/s]

100%|██████████| 113/113 [00:29<00:00,  3.78it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.03it/s]

100%|██████████| 38/38 [00:02<00:00, 18.04it/s]

100%|██████████| 38/38 [00:02<00:00, 18.03it/s]

Epoch 9 : Training Loss: 0.17607, acc: 0.64530, pr_auc: 0.70928, roc_auc: 0.71106, LR: 0.00009, Time elapsed 0.53 mins
[[735 477]
 [385 834]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.76it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.77it/s]

100%|██████████| 113/113 [00:29<00:00,  3.78it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.87it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.89it/s]

100%|██████████| 38/38 [00:02<00:00, 17.87it/s]

Epoch 10 : Training Loss: 0.14124, acc: 0.61375, pr_auc: 0.70309, roc_auc: 0.69302, LR: 0.00009, Time elapsed 0.53 mins
[[750 462]
 [477 742]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.78it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.76it/s]

100%|██████████| 113/113 [00:29<00:00,  3.77it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.68it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.65it/s]

100%|██████████| 38/38 [00:02<00:00, 17.64it/s]

Epoch 11 : Training Loss: 0.11921, acc: 0.61866, pr_auc: 0.68580, roc_auc: 0.68813, LR: 0.00009, Time elapsed 0.54 mins
[[742 470]
 [457 762]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.74it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.74it/s]

100%|██████████| 113/113 [00:30<00:00,  3.76it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.94it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.94it/s]

100%|██████████| 38/38 [00:02<00:00, 17.94it/s]

Epoch 12 : Training Loss: 0.09172, acc: 0.62999, pr_auc: 0.71990, roc_auc: 0.70476, LR: 0.00009, Time elapsed 0.54 mins
[[678 534]
 [365 854]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.71it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.72it/s]

100%|██████████| 113/113 [00:30<00:00,  3.73it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.61it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.26it/s]

100%|██████████| 38/38 [00:02<00:00, 17.32it/s]

Epoch 13 : Training Loss: 0.07809, acc: 0.61376, pr_auc: 0.68835, roc_auc: 0.68430, LR: 0.00008, Time elapsed 0.54 mins
[[754 458]
 [481 738]]


  0%|          | 0/113 [00:00<?, ?it/s]

 33%|███▎      | 37/113 [00:10<00:20,  3.68it/s]

 65%|██████▌   | 74/113 [00:20<00:10,  3.69it/s]

 98%|█████████▊| 111/113 [00:30<00:00,  3.68it/s]

100%|██████████| 113/113 [00:30<00:00,  3.69it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.32it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.12it/s]

100%|██████████| 38/38 [00:02<00:00, 17.18it/s]

Epoch 14 : Training Loss: 0.08042, acc: 0.63545, pr_auc: 0.72518, roc_auc: 0.71060, LR: 0.00008, Time elapsed 0.55 mins
[[731 481]
 [405 814]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.79it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.79it/s]

100%|██████████| 113/113 [00:29<00:00,  3.80it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.32it/s]

100%|██████████| 38/38 [00:02<00:00, 18.12it/s]

100%|██████████| 38/38 [00:02<00:00, 18.14it/s]

Epoch 15 : Training Loss: 0.05942, acc: 0.62772, pr_auc: 0.71458, roc_auc: 0.70350, LR: 0.00008, Time elapsed 0.53 mins
[[759 453]
 [452 767]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.79it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.79it/s]

100%|██████████| 113/113 [00:29<00:00,  3.79it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.05it/s]

100%|██████████| 38/38 [00:02<00:00, 17.97it/s]

100%|██████████| 38/38 [00:02<00:00, 17.97it/s]

Epoch 16 : Training Loss: 0.04764, acc: 0.62602, pr_auc: 0.71132, roc_auc: 0.69845, LR: 0.00008, Time elapsed 0.53 mins
[[734 478]
 [431 788]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.80it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.79it/s]

100%|██████████| 113/113 [00:29<00:00,  3.80it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.28it/s]

100%|██████████| 38/38 [00:02<00:00, 18.22it/s]

100%|██████████| 38/38 [00:02<00:00, 18.22it/s]

Epoch 17 : Training Loss: 0.04134, acc: 0.63591, pr_auc: 0.70205, roc_auc: 0.69496, LR: 0.00007, Time elapsed 0.53 mins
[[753 459]
 [426 793]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.80it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.79it/s]

100%|██████████| 113/113 [00:29<00:00,  3.80it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.17it/s]

100%|██████████| 38/38 [00:02<00:00, 18.11it/s]

100%|██████████| 38/38 [00:02<00:00, 18.10it/s]

Epoch 18 : Training Loss: 0.03498, acc: 0.62848, pr_auc: 0.72464, roc_auc: 0.70778, LR: 0.00007, Time elapsed 0.53 mins
[[732 480]
 [423 796]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.80it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.79it/s]

100%|██████████| 113/113 [00:29<00:00,  3.80it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.16it/s]

100%|██████████| 38/38 [00:02<00:00, 18.17it/s]

100%|██████████| 38/38 [00:02<00:00, 18.16it/s]

Epoch 19 : Training Loss: 0.03574, acc: 0.62273, pr_auc: 0.70117, roc_auc: 0.68550, LR: 0.00007, Time elapsed 0.53 mins
[[729 483]
 [434 785]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.80it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.79it/s]

100%|██████████| 113/113 [00:29<00:00,  3.80it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.07it/s]

100%|██████████| 38/38 [00:02<00:00, 17.95it/s]

100%|██████████| 38/38 [00:02<00:00, 17.96it/s]

Epoch 20 : Training Loss: 0.02939, acc: 0.63355, pr_auc: 0.72359, roc_auc: 0.71206, LR: 0.00007, Time elapsed 0.53 mins
[[795 417]
 [474 745]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.79it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.80it/s]

100%|██████████| 113/113 [00:29<00:00,  3.81it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.23it/s]

100%|██████████| 38/38 [00:02<00:00, 18.17it/s]

100%|██████████| 38/38 [00:02<00:00, 18.17it/s]

Epoch 21 : Training Loss: 0.02719, acc: 0.63225, pr_auc: 0.72151, roc_auc: 0.70727, LR: 0.00006, Time elapsed 0.53 mins
[[765 447]
 [447 772]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.80it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.80it/s]

100%|██████████| 113/113 [00:29<00:00,  3.80it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.13it/s]

100%|██████████| 38/38 [00:02<00:00, 18.08it/s]

100%|██████████| 38/38 [00:02<00:00, 18.08it/s]

Epoch 22 : Training Loss: 0.02030, acc: 0.63130, pr_auc: 0.71934, roc_auc: 0.70422, LR: 0.00006, Time elapsed 0.53 mins
[[710 502]
 [394 825]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.79it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.79it/s]

100%|██████████| 113/113 [00:29<00:00,  3.80it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.21it/s]

100%|██████████| 38/38 [00:02<00:00, 18.10it/s]

100%|██████████| 38/38 [00:02<00:00, 18.10it/s]

Epoch 23 : Training Loss: 0.01870, acc: 0.60889, pr_auc: 0.69674, roc_auc: 0.67687, LR: 0.00006, Time elapsed 0.53 mins
[[774 438]
 [513 706]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.80it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.80it/s]

100%|██████████| 113/113 [00:29<00:00,  3.81it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.07it/s]

100%|██████████| 38/38 [00:02<00:00, 18.06it/s]

100%|██████████| 38/38 [00:02<00:00, 18.05it/s]

Epoch 24 : Training Loss: 0.02125, acc: 0.62726, pr_auc: 0.71701, roc_auc: 0.70153, LR: 0.00005, Time elapsed 0.53 mins
[[736 476]
 [430 789]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.79it/s]

 34%|███▎      | 38/113 [00:20<00:19,  3.79it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.75it/s]

100%|██████████| 113/113 [00:30<00:00,  3.74it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.25it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.43it/s]

100%|██████████| 38/38 [00:02<00:00, 17.38it/s]

Epoch 25 : Training Loss: 0.01316, acc: 0.62154, pr_auc: 0.70574, roc_auc: 0.68879, LR: 0.00005, Time elapsed 0.54 mins
[[746 466]
 [454 765]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.70it/s]

 67%|██████▋   | 76/113 [00:20<00:10,  3.69it/s]

100%|██████████| 113/113 [00:30<00:00,  3.71it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.51it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.27it/s]

100%|██████████| 38/38 [00:02<00:00, 17.33it/s]

Epoch 26 : Training Loss: 0.01234, acc: 0.63229, pr_auc: 0.72014, roc_auc: 0.70822, LR: 0.00005, Time elapsed 0.55 mins
[[782 430]
 [464 755]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.70it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.71it/s]

100%|██████████| 113/113 [00:30<00:00,  3.73it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.61it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.57it/s]

100%|██████████| 38/38 [00:02<00:00, 17.55it/s]

Epoch 27 : Training Loss: 0.01300, acc: 0.63555, pr_auc: 0.72907, roc_auc: 0.71424, LR: 0.00004, Time elapsed 0.54 mins
[[772 440]
 [446 773]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.73it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.73it/s]

100%|██████████| 113/113 [00:30<00:00,  3.74it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.78it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.84it/s]

100%|██████████| 38/38 [00:02<00:00, 17.84it/s]

Epoch 28 : Training Loss: 0.01231, acc: 0.62774, pr_auc: 0.71714, roc_auc: 0.70237, LR: 0.00004, Time elapsed 0.54 mins
[[767 445]
 [460 759]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.82it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.81it/s]

100%|██████████| 113/113 [00:29<00:00,  3.82it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.29it/s]

100%|██████████| 38/38 [00:02<00:00, 18.08it/s]

100%|██████████| 38/38 [00:02<00:00, 18.10it/s]

Epoch 29 : Training Loss: 0.01118, acc: 0.63143, pr_auc: 0.71749, roc_auc: 0.70091, LR: 0.00004, Time elapsed 0.53 mins
[[765 447]
 [449 770]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.81it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.80it/s]

100%|██████████| 113/113 [00:29<00:00,  3.81it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.21it/s]

100%|██████████| 38/38 [00:02<00:00, 18.05it/s]

100%|██████████| 38/38 [00:02<00:00, 18.06it/s]

Epoch 30 : Training Loss: 0.01162, acc: 0.62567, pr_auc: 0.71516, roc_auc: 0.69779, LR: 0.00004, Time elapsed 0.53 mins
[[760 452]
 [458 761]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.80it/s]

 68%|██████▊   | 77/113 [00:20<00:09,  3.81it/s]

100%|██████████| 113/113 [00:29<00:00,  3.82it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.13it/s]

100%|██████████| 38/38 [00:02<00:00, 18.02it/s]

100%|██████████| 38/38 [00:02<00:00, 18.03it/s]

Epoch 31 : Training Loss: 0.01113, acc: 0.62777, pr_auc: 0.71196, roc_auc: 0.69706, LR: 0.00003, Time elapsed 0.53 mins
[[779 433]
 [472 747]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.79it/s]

 68%|██████▊   | 77/113 [00:20<00:09,  3.80it/s]

100%|██████████| 113/113 [00:29<00:00,  3.81it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.03it/s]

100%|██████████| 38/38 [00:02<00:00, 18.11it/s]

100%|██████████| 38/38 [00:02<00:00, 18.09it/s]

Epoch 32 : Training Loss: 0.01213, acc: 0.62653, pr_auc: 0.71588, roc_auc: 0.70157, LR: 0.00003, Time elapsed 0.53 mins
[[776 436]
 [472 747]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.80it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.80it/s]

100%|██████████| 113/113 [00:29<00:00,  3.81it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.17it/s]

100%|██████████| 38/38 [00:02<00:00, 18.12it/s]

100%|██████████| 38/38 [00:02<00:00, 18.12it/s]

Epoch 33 : Training Loss: 0.01084, acc: 0.61251, pr_auc: 0.70123, roc_auc: 0.68900, LR: 0.00003, Time elapsed 0.53 mins
[[745 467]
 [475 744]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.81it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.80it/s]

100%|██████████| 113/113 [00:29<00:00,  3.81it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.06it/s]

100%|██████████| 38/38 [00:02<00:00, 17.97it/s]

100%|██████████| 38/38 [00:02<00:00, 17.98it/s]

Epoch 34 : Training Loss: 0.01134, acc: 0.61910, pr_auc: 0.70516, roc_auc: 0.68882, LR: 0.00002, Time elapsed 0.53 mins
[[755 457]
 [469 750]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.81it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.81it/s]

100%|██████████| 113/113 [00:29<00:00,  3.82it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.17it/s]

100%|██████████| 38/38 [00:02<00:00, 18.07it/s]

100%|██████████| 38/38 [00:02<00:00, 18.08it/s]

Epoch 35 : Training Loss: 0.01077, acc: 0.62316, pr_auc: 0.70557, roc_auc: 0.68852, LR: 0.00002, Time elapsed 0.53 mins
[[740 472]
 [444 775]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.81it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.81it/s]

100%|██████████| 113/113 [00:29<00:00,  3.82it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.28it/s]

100%|██████████| 38/38 [00:02<00:00, 18.11it/s]

100%|██████████| 38/38 [00:02<00:00, 18.12it/s]

Epoch 36 : Training Loss: 0.01064, acc: 0.62645, pr_auc: 0.70772, roc_auc: 0.69082, LR: 0.00002, Time elapsed 0.53 mins
[[744 468]
 [440 779]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.82it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.82it/s]

100%|██████████| 113/113 [00:29<00:00,  3.83it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.28it/s]

100%|██████████| 38/38 [00:02<00:00, 18.13it/s]

100%|██████████| 38/38 [00:02<00:00, 18.14it/s]

Epoch 37 : Training Loss: 0.01070, acc: 0.63057, pr_auc: 0.72095, roc_auc: 0.70457, LR: 0.00002, Time elapsed 0.53 mins
[[750 462]
 [436 783]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.82it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.82it/s]

100%|██████████| 113/113 [00:29<00:00,  3.83it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.33it/s]

100%|██████████| 38/38 [00:02<00:00, 18.07it/s]

100%|██████████| 38/38 [00:02<00:00, 18.10it/s]

Epoch 38 : Training Loss: 0.01046, acc: 0.63430, pr_auc: 0.72105, roc_auc: 0.70477, LR: 0.00001, Time elapsed 0.53 mins
[[767 445]
 [444 775]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.82it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.80it/s]

100%|██████████| 113/113 [00:29<00:00,  3.81it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.09it/s]

100%|██████████| 38/38 [00:02<00:00, 17.99it/s]

100%|██████████| 38/38 [00:02<00:00, 17.99it/s]

Epoch 39 : Training Loss: 0.01040, acc: 0.63307, pr_auc: 0.72037, roc_auc: 0.70418, LR: 0.00001, Time elapsed 0.53 mins
[[766 446]
 [446 773]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.78it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.79it/s]

100%|██████████| 113/113 [00:29<00:00,  3.80it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.22it/s]

100%|██████████| 38/38 [00:02<00:00, 18.07it/s]

100%|██████████| 38/38 [00:02<00:00, 18.08it/s]

Epoch 40 : Training Loss: 0.01036, acc: 0.63431, pr_auc: 0.71818, roc_auc: 0.70291, LR: 0.00001, Time elapsed 0.53 mins
[[772 440]
 [449 770]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.77it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.74it/s]

100%|██████████| 113/113 [00:30<00:00,  3.76it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.73it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.68it/s]

100%|██████████| 38/38 [00:02<00:00, 17.71it/s]

Epoch 41 : Training Loss: 0.01065, acc: 0.63679, pr_auc: 0.72119, roc_auc: 0.70545, LR: 0.00001, Time elapsed 0.54 mins
[[776 436]
 [447 772]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.72it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.72it/s]

100%|██████████| 113/113 [00:30<00:00,  3.74it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 47%|████▋     | 18/38 [00:01<00:01, 17.69it/s]

 95%|█████████▍| 36/38 [00:02<00:00, 17.63it/s]

100%|██████████| 38/38 [00:02<00:00, 17.62it/s]

Epoch 42 : Training Loss: 0.01055, acc: 0.63349, pr_auc: 0.72213, roc_auc: 0.70545, LR: 0.00001, Time elapsed 0.54 mins
[[772 440]
 [451 768]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:20,  3.74it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.77it/s]

100%|██████████| 113/113 [00:29<00:00,  3.79it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.10it/s]

100%|██████████| 38/38 [00:02<00:00, 18.06it/s]

100%|██████████| 38/38 [00:02<00:00, 18.05it/s]

Epoch 43 : Training Loss: 0.01078, acc: 0.63101, pr_auc: 0.71900, roc_auc: 0.70218, LR: 0.00001, Time elapsed 0.53 mins
[[761 451]
 [446 773]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.79it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.80it/s]

100%|██████████| 113/113 [00:29<00:00,  3.81it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.01it/s]

100%|██████████| 38/38 [00:02<00:00, 17.90it/s]

100%|██████████| 38/38 [00:02<00:00, 17.90it/s]

Epoch 44 : Training Loss: 0.01058, acc: 0.62856, pr_auc: 0.71827, roc_auc: 0.70216, LR: 0.00000, Time elapsed 0.53 mins
[[767 445]
 [458 761]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.79it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.79it/s]

100%|██████████| 113/113 [00:29<00:00,  3.81it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 17.99it/s]

 97%|█████████▋| 37/38 [00:02<00:00, 17.97it/s]

100%|██████████| 38/38 [00:02<00:00, 17.96it/s]

Epoch 45 : Training Loss: 0.01055, acc: 0.62651, pr_auc: 0.71877, roc_auc: 0.70241, LR: 0.00000, Time elapsed 0.53 mins
[[767 445]
 [463 756]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.78it/s]

 67%|██████▋   | 76/113 [00:20<00:09,  3.79it/s]

100%|██████████| 113/113 [00:29<00:00,  3.80it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.19it/s]

100%|██████████| 38/38 [00:02<00:00, 18.08it/s]

100%|██████████| 38/38 [00:02<00:00, 18.08it/s]

Epoch 46 : Training Loss: 0.01088, acc: 0.62651, pr_auc: 0.71856, roc_auc: 0.70204, LR: 0.00000, Time elapsed 0.53 mins
[[768 444]
 [464 755]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.81it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.81it/s]

100%|██████████| 113/113 [00:29<00:00,  3.82it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.09it/s]

100%|██████████| 38/38 [00:02<00:00, 18.24it/s]

100%|██████████| 38/38 [00:02<00:00, 18.20it/s]

Epoch 47 : Training Loss: 0.01079, acc: 0.62733, pr_auc: 0.71819, roc_auc: 0.70182, LR: 0.00000, Time elapsed 0.53 mins
[[768 444]
 [462 757]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.81it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.81it/s]

100%|██████████| 113/113 [00:29<00:00,  3.82it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.21it/s]

100%|██████████| 38/38 [00:02<00:00, 18.13it/s]

100%|██████████| 38/38 [00:02<00:00, 18.13it/s]

Epoch 48 : Training Loss: 0.01068, acc: 0.63021, pr_auc: 0.71885, roc_auc: 0.70245, LR: 0.00000, Time elapsed 0.53 mins
[[771 441]
 [458 761]]


  0%|          | 0/113 [00:00<?, ?it/s]

 34%|███▎      | 38/113 [00:10<00:19,  3.80it/s]

 68%|██████▊   | 77/113 [00:20<00:09,  3.80it/s]

100%|██████████| 113/113 [00:29<00:00,  3.81it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.33it/s]

100%|██████████| 38/38 [00:02<00:00, 18.15it/s]

100%|██████████| 38/38 [00:02<00:00, 18.17it/s]

Epoch 49 : Training Loss: 0.01061, acc: 0.63021, pr_auc: 0.71889, roc_auc: 0.70249, LR: 0.00000, Time elapsed 0.53 mins
[[771 441]
 [458 761]]


  0%|          | 0/113 [00:00<?, ?it/s]

 35%|███▍      | 39/113 [00:10<00:19,  3.82it/s]

 69%|██████▉   | 78/113 [00:20<00:09,  3.82it/s]

100%|██████████| 113/113 [00:29<00:00,  3.83it/s]

  0%|          | 0/38 [00:00<?, ?it/s]

 50%|█████     | 19/38 [00:01<00:01, 18.13it/s]

100%|██████████| 38/38 [00:02<00:00, 18.12it/s]

100%|██████████| 38/38 [00:02<00:00, 18.12it/s]

Epoch 50 : Training Loss: 0.01043, acc: 0.63144, pr_auc: 0.71932, roc_auc: 0.70282, LR: 0.00000, Time elapsed 0.53 mins
[[771 441]
 [455 764]]
***************************Test************************


  0%|          | 0/37 [00:00<?, ?it/s]

 51%|█████▏    | 19/37 [00:01<00:00, 18.02it/s]

100%|██████████| 37/37 [00:02<00:00, 18.05it/s]

***************************Test results************************
Test Evaluation: acc: 0.61498, pr_auc: 0.70104, roc_auc: 0.68130
[[896 273]
 [632 546]]
model save in checkpoints/epoch5_acc_0.61498_pr_0.70104_roc_0.68130.pth
Done!!!!!


## Brief note :

Last model save happens at epoch 5, which is a clear sign of overfitting on the training dataset, it memorizes the training set rather than generalizing across subjects.

The metric results on the test set, with seed 3407, are : acc: 0.61498, pr_auc: 0.70104, roc_auc: 0.68130.

While in the paper, it is reported : acc: 0.6370 ± 0.0151, pr_auc: 0.7139 ± 0.0088, roc_auc: 0.6988 ± 0.0068.

So the results are clearly lower than expected.